tests/core/contracts/test_contract_class_construction.py

```diff
@@ -42,6 +42,20 @@ def test_abi_as_json_string(w3, math_contract_abi, some_address):
     assert math.abi == math_contract_abi
 
 
+def test_contract_init_with_w3_function_name(
+    w3,
+    function_name_tester_contract_abi,
+    function_name_tester_contract,  # this is the fixture that deployed the contract
```

Comment:
We can remove these comments from our conversation. We can probably still add some more succinct notes in the body of the test for a reader. Maybe something like:

```py
# test `w3` function name does not throw when creating the contract factory
contract_factory = w3.eth.contract(abi=function_name_tester_contract_abi) 
    
# re-instantiate the contract and test contract interaction
contract = contract_factory(function_name_tester_contract.address)
assert contract.functions.w3().call() is True
```